core/runtime/binaryen/runtime_external_interface.cpp

```diff
@@ -149,119 +264,11 @@ namespace kagome::runtime::binaryen {
-
-      CALL_HOST_API_FUNC(ext_allocator_malloc_version_1);
-      CALL_HOST_API_FUNC(ext_allocator_free_version_1);
-
-      // -------------------------- storage functions --------------------------
-
-      CALL_HOST_API_FUNC(ext_storage_set_version_1);
-      CALL_HOST_API_FUNC(ext_storage_get_version_1);
-      CALL_HOST_API_FUNC(ext_storage_clear_version_1);
-      CALL_HOST_API_FUNC(ext_storage_exists_version_1);
-      CALL_HOST_API_FUNC(ext_storage_read_version_1);
-      CALL_HOST_API_FUNC(ext_storage_clear_prefix_version_1);
-      CALL_HOST_API_FUNC(ext_storage_clear_prefix_version_2);
-      CALL_HOST_API_FUNC(ext_storage_root_version_1);
-      CALL_HOST_API_FUNC(ext_storage_root_version_2);
-      CALL_HOST_API_FUNC(ext_storage_changes_root_version_1);
-      CALL_HOST_API_FUNC(ext_storage_next_key_version_1);
-      CALL_HOST_API_FUNC(ext_storage_append_version_1);
-      CALL_HOST_API_FUNC(ext_storage_start_transaction_version_1);
-      CALL_HOST_API_FUNC(ext_storage_commit_transaction_version_1);
-      CALL_HOST_API_FUNC(ext_storage_rollback_transaction_version_1);
-      CALL_HOST_API_FUNC(ext_trie_blake2_256_root_version_1);
-      CALL_HOST_API_FUNC(ext_trie_blake2_256_ordered_root_version_1);
-      CALL_HOST_API_FUNC(ext_trie_blake2_256_ordered_root_version_2);
-      CALL_HOST_API_FUNC(ext_misc_print_hex_version_1);
-      CALL_HOST_API_FUNC(ext_misc_print_num_version_1);
-      CALL_HOST_API_FUNC(ext_misc_print_utf8_version_1);
-      CALL_HOST_API_FUNC(ext_misc_runtime_version_version_1);
-
-      // ------------------------- Offchain extension --------------------------
-
-      CALL_HOST_API_FUNC(ext_offchain_is_validator_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_submit_transaction_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_network_state_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_timestamp_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_sleep_until_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_random_seed_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_local_storage_set_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_local_storage_clear_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_local_storage_compare_and_set_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_local_storage_get_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_http_request_start_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_http_request_add_header_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_http_request_write_body_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_http_response_wait_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_http_response_headers_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_http_response_read_body_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_set_authorized_nodes_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_index_set_version_1);
-      CALL_HOST_API_FUNC(ext_offchain_index_clear_version_1);
+      auto it = imports_.find(import->base.c_str());
```

Comment:
Unnecessary allocation
```suggestion
      auto it = imports_.find(import->base);
```